### ISSUE TITLE
NC15 - Unhandled exception occ files:scan 

### ISSUE THREAD

**Issue by guzi42** [nextcloud/server#14843]
### Steps to reproduce
1.occ files:scan --all
2.
3.

### Expected behaviour
should complete without errors

### Actual behaviour
Starting scan for user 34 out of 379 (someuser)
An unhandled exception has been thrown:
TypeError: Argument 1 passed to OC\OCS\DiscoveryService::discover() must be of the type string, null given, called in /var/www/owncloud/apps/files_sharing/lib/External/Manager.php on line 382 and defined in /var/www/owncloud/lib/private/OCS/DiscoveryService.php:65
Stack trace:
#0 /var/www/owncloud/apps/files_sharing/lib/External/Manager.php(382): OC\OCS\DiscoveryService->discover(NULL, 'FEDERATED_SHARI...')
#1 /var/www/owncloud/apps/files_sharing/lib/External/Manager.php(527): OCA\Files_Sharing\External\Manager->sendFeedbackToRemote(NULL, NULL, NULL, 'decline')
#2 /var/www/owncloud/apps/files_sharing/lib/External/Storage.php(218): OCA\Files_Sharing\External\Manager->removeShare('/files/Global E...')
#3 [internal function]: OCA\Files_Sharing\External\Storage->checkStorageAvailability()
#4 /var/www/owncloud/lib/private/Files/Storage/Wrapper/Wrapper.php(506): call_user_func_array(Array, Array)
#5 [internal function]: OC\Files\Storage\Wrapper\Wrapper->__call('checkStorageAva...', Array)
#6 /var/www/owncloud/lib/private/Files/Storage/Wrapper/Wrapper.php(506): call_user_func_array(Array, Array)
#7 /var/www/owncloud/apps/files_sharing/lib/External/Scanner.php(89): OC\Files\Storage\Wrapper\Wrapper->__call('checkStorageAva...', Array)
#8 /var/www/owncloud/apps/files_sharing/lib/External/Scanner.php(44): OCA\Files_Sharing\External\Scanner->scanAll()
#9 /var/www/owncloud/lib/private/Files/Utils/Scanner.php(245): OCA\Files_Sharing\External\Scanner->scan('', true, 3)
#10 /var/www/owncloud/apps/files/lib/Command/Scan.php(145): OC\Files\Utils\Scanner->scan('/Didiot', true, NULL)
#11 /var/www/owncloud/apps/files/lib/Command/Scan.php(201): OCA\Files\Command\Scan->scanFiles('someuser', '/someuser', Object(Symfony\Component\Console\Output\ConsoleOutput), false, true, false)
#12 /var/www/owncloud/3rdparty/symfony/console/Command/Command.php(255): OCA\Files\Command\Scan->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /var/www/owncloud/core/Command/Base.php(166): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /var/www/owncloud/3rdparty/symfony/console/Application.php(946): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /var/www/owncloud/3rdparty/symfony/console/Application.php(248): Symfony\Component\Console\Application->doRunCommand(Object(OCA\Files\Command\Scan), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /var/www/owncloud/3rdparty/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/owncloud/lib/private/Console/Application.php(213): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/owncloud/console.php(96): OC\Console\Application->run()
#19 /var/www/owncloud/occ(11): require_once('/var/www/ownclo...')
#20 {main}root@neksowncloud:/media/ownclouddata/Didiot# 

### Server configuration

**OS:** Ubuntu 18.04.2 LTS

**web server:** apache2     2.4.29-1ubuntu4.5 

**Database:**  mysql-server-5.7                      5.7.25-0ubuntu0.18.04.2

**PHP version:** 7.2

**Nextcloud version:** 15.0.5

**Updated from an older Nextcloud/ownCloud or fresh install:** from ownCloud 10 via nextcloud  12/13/14

**Where did you install Nextcloud from:**download from web installer

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.

No errors have been found.

</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

### Logs
#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by ChristophWurst** [comment]
Fixed for 16 and requires a backport to 15: https://github.com/nextcloud/server/pull/13790#issuecomment-480238905

**Comment by MorrisJobke** [comment]
fixed via #14968
